virtcontainers/sandbox.go

```diff
@@ -1583,33 +1583,42 @@ func (s *Sandbox) setSandboxState(state types.StateString) error {
 	return nil
 }
 
-// getAndSetSandboxBlockIndex retrieves sandbox block index and increments it for
-// subsequent accesses. This index is used to maintain the index at which a
-// block device is assigned to a container in the sandbox.
+const maxBlockIndex = 65535
+
+// getAndSetSandboxBlockIndex retrieves an unused sandbox block index from
+// the BlockIndexMap and marks it as used. This index is used to maintain the
+// index at which a block device is assigned to a container in the sandbox.
 func (s *Sandbox) getAndSetSandboxBlockIndex() (int, error) {
-	currentIndex := s.state.BlockIndex
 	var err error
+	currentIndex := -1
+	for i := 0; i < maxBlockIndex; i++ {
+		if _, ok := s.state.BlockIndexMap[i]; !ok {
+			currentIndex = i
```

Comment:
so, if `BlockIndexMap` is empty , `currentIndex` will be set to `0`, but in the next lines you are comparing `currentIndex == -1` ? is that correct?

Reply:
thanks @devimc for the questions . This loop aims to find the first unused index from the index map. If `BlockIndexMap` is empty, then `currentIndex` will be the first location of bitmap with value **0**. If `map[0]` to `map[maxBlockIndex-1]` are all marked as used, it means that there is no more index in the bitmap can be reused, and `currentIndex` will not be assigned but leave with **-1**. Then we  throw an error in such condition.

Reply:
thanks @darfux , that make sense